From 0e10080b6af411f9c6a89827fd900cf6459ac238 Mon Sep 17 00:00:00 2001
From: Praveen Diwakar <praveen.diwakar@intel.com>
Date: Tue, 28 Apr 2015 20:50:55 +0530
Subject: [PATCH] Revert "Audio: CHT-CR: Workaround to fix spurious headset
 interrupt at boot"

This reverts commit 612e0452dcb90ee615e5ad1ed94d3706def1ccdf.
As this patch is workaround submitted earlier, its no more required
as clean fix got merged already.

Change-Id: I4f15d23a69b281b45d1b0f75e34d050703c9f587
Tracked-On: https://jira01.devtools.intel.com/browse/GMINL-8741
Signed-off-by: Praveen Diwakar <praveen.diwakar@intel.com>
---
 sound/soc/codecs/tlv320aic31xx.c       | 10 +++-------
 sound/soc/intel/board/cht_cr_aic3100.c | 17 +++--------------
 2 files changed, 6 insertions(+), 21 deletions(-)

diff --git a/sound/soc/codecs/tlv320aic31xx.c b/sound/soc/codecs/tlv320aic31xx.c
index 579cd45d3ecc..ce8cc3a0d374 100644
--- a/sound/soc/codecs/tlv320aic31xx.c
+++ b/sound/soc/codecs/tlv320aic31xx.c
@@ -1327,6 +1327,9 @@ static int aic31xx_codec_probe(struct snd_soc_codec *codec)
 			AIC31XX_HSPLUGDET_MASK,
 			AIC31XX_HSPLUGDET_MASK);
 	snd_soc_update_bits(codec, AIC31XX_INT1CTRL,
+			AIC31XX_HSPLUGDET_MASK,
+			AIC31XX_HSPLUGDET_MASK);
+	snd_soc_update_bits(codec, AIC31XX_INT1CTRL,
 			AIC31XX_BUTTONPRESSDET_MASK,
 			AIC31XX_BUTTONPRESSDET_MASK);
 	/* Program codec to use internal clock */
@@ -1366,13 +1369,6 @@ static int aic31xx_codec_probe(struct snd_soc_codec *codec)
 	aic31xx_add_widgets(codec);
 	dev_dbg(codec->dev, "%d, %s, Firmware test\n", __LINE__, __func__);
 
-	snd_soc_update_bits(codec, AIC31XX_INT1CTRL,
-			AIC31XX_HSPLUGDET_MASK,
-			AIC31XX_HSPLUGDET_MASK);
-	msleep(250);
-	snd_soc_read(codec, AIC31XX_INTRDACFLAG);
-	snd_soc_read(codec, AIC31XX_INTRFLAG);
-
 	return ret;
 }
 
diff --git a/sound/soc/intel/board/cht_cr_aic3100.c b/sound/soc/intel/board/cht_cr_aic3100.c
index 6e9d000aef35..be28bfad68b1 100644
--- a/sound/soc/intel/board/cht_cr_aic3100.c
+++ b/sound/soc/intel/board/cht_cr_aic3100.c
@@ -41,7 +41,7 @@
 
 #define CHT_PLAT_CLK_3_HZ	25000000
 
-#define CHT_INTR_DEBOUNCE               2000
+#define CHT_INTR_DEBOUNCE               0
 #define CHT_HS_INSERT_DET_DELAY         100
 #define CHT_HS_REMOVE_DET_DELAY         100
 #define CHT_BUTTON_DET_DELAY            50
@@ -168,7 +168,6 @@ static int cht_check_jack_type(struct snd_soc_jack *jack,
 static int cht_hs_detection(void *data)
 {
 	int status, jack_type = 0;
-	static int boot_debounce;
 	int ret, val, instantaneous;
 	struct snd_soc_jack_gpio *gpio = &hs_gpio;
 	struct snd_soc_jack *jack = gpio->jack;
@@ -188,15 +187,7 @@ static int cht_hs_detection(void *data)
 	pr_debug("Enter:%s Page0/44=0x%x, Page0/46=0x%x", __func__,
 						val, instantaneous);
 
-	/* Debounce time for boot is 2 second due to noise from hardarware jack
-	   which generates spurious interrupt. After boot debounce time is again
-	   set to 100ms */
-	if (!boot_debounce) {
-		gpio->debounce_time = 100;
-		boot_debounce = 1;
-	}
-
-	if ((!jack->status) && (val & AIC31XX_HSPLUG_MASK)) {
+	if (!jack->status) {
 		ctx->hs_det_retry = CHT_HS_DET_RETRY_COUNT;
 		ret = schedule_delayed_work(&ctx->hs_insert_work,
 				msecs_to_jiffies(ctx->hs_insert_det_delay));
@@ -214,7 +205,7 @@ static int cht_hs_detection(void *data)
 		/* jd status low indicates accessory has been disconnected.
 		 * However, confirm the removal in the delayed work
 		 */
-		if ((!status) && (val & AIC31XX_HSPLUG_MASK)) {
+		if (!status) {
 			/* Do not process button events while we make sure
 			 * accessory is disconnected
 			 */
@@ -709,7 +700,6 @@ static int cht_audio_init(struct snd_soc_pcm_runtime *runtime)
 		return ret;
 	}
 
-
 	ret = snd_soc_add_card_controls(card, cht_mc_controls,
 					ARRAY_SIZE(cht_mc_controls));
 	if (ret) {
@@ -722,7 +712,6 @@ static int cht_audio_init(struct snd_soc_pcm_runtime *runtime)
 		pr_err("unable to sync dapm\n");
 		return ret;
 	}
-
 	return ret;
 }
 
-- 
1.9.1

